tests/security_test.go

```diff
@@ -295,3 +297,93 @@ func runTestCase(tc securityTestCase) error {
 
 	return err
 }
+
+func TestEnvBasedSecrets(t *testing.T) {
+	err := runSecretTest("env-based-secret", "", "secretKey", "set to the key 'secretKey' in secret 'env-based-secret'")
+
+	if err != nil {
+		t.Error(err.Error())
+	}
+}
+
+func TestFileBasedSecrets(t *testing.T) {
+	err := runSecretTest("file-based-secret", "/mnt/secrets", "", "/mnt/secrets from file-based-secret-volume")
+
+	if err != nil {
+		t.Error(err.Error())
+	}
+}
+
+func runSecretTest(secretName string, secretPath string, secretKey string, expectedSecret string) error {
+	spark := utils.SparkOperatorInstallation{}
+	err := spark.InstallSparkOperator()
+	defer spark.CleanUp()
+
+	if err != nil {
+		return err
+	}
+
+	client, err := utils.GetK8sClientSet()
```

Comment:
After calling `spark.InstallSparkOperator()`, the `spark` struct will already have a client set ready to use as `spark.K8sClients` which you can pass to `CreateSecret` function

Reply:
Thanks @rpalaznik for pointing it out. Changed it.